### PR TITLE
DM-44822: Add CLI commands to list and delete Cassandra instances

### PR DESCRIPTION
**Pull request by andy-slac**
ApdbCassandra adds two class methods to list keyspaces in the cluster and delete keyspaces. Two CLI sub-commands also added that use these new methods.

### CODE REVIEW

File: python/lsst/dax/apdb/cassandra/apdbCassandra.py [hunk 1]
```diff
@@ -592,6 +592,60 @@ def init_database(
...
         return config
 
+    @classmethod
+    def list_databases(cls, host: str) -> Iterable[str]:
+        """Return the list of keyspaces with APDB databases.
+
+        Parameters
+        ----------
+        host : `str`
+            Name of one of the nosts in Cassandra cluster.
```
Comment (isullivan) [inline python/lsst/dax/apdb/cassandra/apdbCassandra.py:602]: "nosts" - > "hosts"

File: python/lsst/dax/apdb/cassandra/apdbCassandra.py [hunk 2]
```diff
@@ -592,6 +592,60 @@ def init_database(
...
+        keyspaces = [row[0] for row in result.all()]
+
+        # Need explicit shutdown.
+        cluster.shutdown()
+
+        return keyspaces
+
+    @classmethod
+    def delete_database(cls, host: str, keyspace: str) -> None:
+        """Return the list of keyspaces with APDB databases.
```
Comment (isullivan) [inline python/lsst/dax/apdb/cassandra/apdbCassandra.py:627]: Fix docstring

File: python/lsst/dax/apdb/cassandra/apdbCassandra.py [hunk 3]
```diff
@@ -592,6 +592,60 @@ def init_database(
...
+
+        Parameters
+        ----------
+        host : `str`
+            Name of one of the nosts in Cassandra cluster.
```
Comment (isullivan) [inline python/lsst/dax/apdb/cassandra/apdbCassandra.py:632]: "nosts" -> "hosts"

File: python/lsst/dax/apdb/cassandra/apdbCassandra.py [hunk 4]
```diff
@@ -592,6 +592,60 @@ def init_database(
...
+        """
+        # For DbAuth we need to use database name "*" to try to match any
+        # database.
+        config = ApdbCassandraConfig(contact_points=[host], keyspace="*")
+        cluster, session = cls._make_session(config)
+
+        # Dropping a keyspace is usually a long operation, set a very long
+        # timeout.
+        query = f"DROP KEYSPACE {quote_id(keyspace)}"
+        session.execute(query, timeout=3600)
```
Comment (isullivan) [inline python/lsst/dax/apdb/cassandra/apdbCassandra.py:644]: Consider making the timeout a keyword parameter, with this default value.

File: python/lsst/dax/apdb/cli/apdb_cli.py [hunk 5]
```diff
@@ -78,6 +80,28 @@ def _create_cassandra_subcommand(subparsers: argparse._SubParsersAction) -> None
...
+    parser = subparsers.add_parser("list-cassandra", help="List APDB instances in Cassandra cluster.")
+    parser.add_argument("host", help="One of the host names for Cassandra cluster.")
+    parser.set_defaults(method=scripts.list_cassandra)
+
+
+def _delete_cassandra_subcommand(subparsers: argparse._SubParsersAction) -> None:
+    parser = subparsers.add_parser("delete-cassandra", help="Delete APDB instance from Cassandra cluster.")
+    parser.add_argument("host", help="One of the host names for Cassandra cluster.")
+    parser.add_argument(
+        "keyspace", help="Cassandra keyspace name for APDB tables, will be created if does not exist."
```
Comment (isullivan) [inline python/lsst/dax/apdb/cli/apdb_cli.py:93]: Is it true that the keyspace will be created if it doesn't exist? I would rather see the delete function fail in that case.

**Review by isullivan — APPROVED** [review]
A few docstring cleanups requested, and a question about the behavior of `delete_cassandra` if the keyspace doesn't exist.

File: python/lsst/dax/apdb/cli/apdb_cli.py (hunk 5 above)
Comment (andy-slac) [reply python/lsst/dax/apdb/cli/apdb_cli.py:93]: My fault, forgot to update it after copy/paste. It is indeed an error if keyspace does not exist, an exception will be raised in that case.

**Pull request merged**